Gems/PhysX/Code/Source/HeightfieldColliderComponent.cpp

```diff
@@ -62,22 +63,83 @@ namespace PhysX
     {
     }
 
+
+    void HeightfieldColliderComponent::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset)
+    {
+        if (asset == m_bakedHeightfieldAsset)
+        {
+            m_bakedHeightfieldAsset = asset;
```

Comment:
Always tricks me when I see this :) For users that don't know about how Asset class works this is an odd thing to do.